### PR TITLE
[WIP] TravisCI: fix and improve MacOS build

### PR DESCRIPTION
**Pull request by TSC21**
This is a continuation of the work being done for bringing `gstreamer` to the MacOSX build and make it pass. Adds some new build pipes so to cross test in different MacOSX versions.

### CODE REVIEW

**Comment by TSC21** [comment]
@julianoes it already finds gstreamer on the MacOS build but still fails because of the protobuf issue. Maybe the bottles weren't updated yet.

File: .travis.yml [hunk 1]
```diff
@@ -79,9 +79,25 @@ matrix:
           env:
             - PX4_DOCKER_REPO=px4io/px4-dev-simulation-bionic:2019-10-04
             - BUILD="source ./scripts/validate_sdf.bash"
-        - name: MacOSX 10.14 (Xcode 11.0) build with Gazebo 9
+        - name: MacOSX Sierra (Xcode 9.2) build with Gazebo 7
```
Comment (julianoes) [inline .travis.yml:82]: I would drop this version. Supporting 2 and with Catalina 3 is enough.

File: CMakeLists.txt [hunk 2]
```diff
@@ -37,16 +40,30 @@ find_program(px4 REQUIRED)
...
     else()
-      find_package(Qt5Widgets REQUIRED)
-      find_package(Qt5Core REQUIRED)
+      # In order to have Qt5 found in MacOS/MacOSX, the Qt5 path needs to be
+      # added to the CMake prefix path
+      if(APPLE)
+        execute_process(COMMAND export Qt5_DIR=$(brew --prefix qt5))
+        execute_process(COMMAND brew --prefix qt5
+                        ERROR_QUIET
+                        OUTPUT_VARIABLE QT5_PREFIX_PATH
```
Comment (julianoes) [inline CMakeLists.txt:61]: Ah, actually this one is nicer than my solution.

File: .travis.yml [hunk 3]
```diff
@@ -90,40 +106,49 @@ matrix:
...
 
 script:
   - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then
-    mkdir Build;
-    cd Build && cmake -DENABLE_UNIT_TESTS=On .. && make -j$(sysctl -n hw.physicalcpu) -l$(sysctl -n hw.physicalcpu); make test;
-    else docker run -it --rm -w ${TRAVIS_BUILD_DIR} --env=CCACHE_DIR="${CCACHE_DIR}" --volume=${CCACHE_DIR}:${CCACHE_DIR}:rw --volume=${TRAVIS_BUILD_DIR}:${TRAVIS_BUILD_DIR}:rw ${PX4_DOCKER_REPO} /bin/bash -e -c "${BUILD}";
+      mkdir Build;
+      cd Build && cmake -DENABLE_UNIT_TESTS=On .. && make -j$(sysctl -n hw.physicalcpu) -l$(sysctl -n hw.physicalcpu); make test;
+    else
+      docker run -it --rm -w ${TRAVIS_BUILD_DIR} --env=CCACHE_DIR="${CCACHE_DIR}" --volume=${CCACHE_DIR}:${CCACHE_DIR}:rw --volume=${TRAVIS_BUILD_DIR}:${TRAVIS_BUILD_DIR}:rw ${PX4_DOCKER_REPO} /bin/bash -e -c "${BUILD}";
```
Comment (julianoes) [inline .travis.yml:153]: @TSC21 what do we use docker for here?

Comment (TSC21) [reply .travis.yml:153]: To build on the Linux distros.

Comment (julianoes) [reply .travis.yml:153]: But that's inside the `if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then`?

Comment (julianoes) [reply .travis.yml:153]: Oh, there is the else :see_no_evil:

**Comment by TSC21** [comment]
Superseded by https://github.com/PX4/sitl_gazebo/pull/354

File: .travis.yml (hunk 1 above)
Comment (TSC21) [reply .travis.yml:82]: I am adding this one so we can have a build pipe for Gazebo 7 also in MacOSX.

**Pull request closed without merging**